polus-k-means-clustering-plugin/src/main.py

```diff
@@ -0,0 +1,244 @@
+    parser.add_argument('--numofclus', dest='numofclus', type=int,
+                        help='Number of clusters', required=False)
+    parser.add_argument('--outdir', dest='outdir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()  
+    
+    #Path to csvfile directory
+    inpdir = args.inpdir
+    logger.info('inpdir = {}'.format(inpdir))
+    
+    #Determine k-value using different methods
+    determinek = args.determinek
+    logger.info('determinek = {}'.format(determinek))
+    
+    #minimum range value to consider for determining k-value
+    minimumrange = args.minimumrange
+    logger.info('minimumrange = {}'.format(minimumrange))
+    
+    #maximum range value to consider for determining k-value
+    maximumrange = args.maximumrange
+    logger.info('maximumrange = {}'.format(maximumrange))
+    
+    #k-value for clustering using K-Means
+    numofclus = args.numofclus
+    logger.info('numofclus = {}'.format(numofclus))
+    
+    #Path to save output csvfiles
+    outdir = args.outdir
+    logger.info('outdir = {}'.format(outdir))
+    
+        
+    #Get list of .csv files in the directory including sub folders for clustering
+    inputcsv = list_file(inpdir)
+    if not inputcsv:
+            raise ValueError('No .csv files found.')
+            
+    #Dictionary of methods to determine k-value
+    FEAT = {'Elbow': elbow,
+            'CalinskiHarabasz': calinski_harabasz,
+            'DaviesBouldin': davies_bouldin}
+       
+    for inpfile in inputcsv:
+        #Get the full path
+        split_file = os.path.normpath(inpfile)
+        
+        #split to get only the filename
+        inpfilename = os.path.split(split_file)
+        file_name_csv = inpfilename[-1]
+        file_name,file_name1 = file_name_csv.split('.', 1)
+        logger.info('Started reading the file ' + file_name)
+        
+        #Read the csv files using pandas
+        data= pd.read_csv(inpfile)
+        
+        #Check whether any one of the method is selected for determining k-value
+        if not determinek and not numofclus:
+            raise ValueError('Select method to determine k-value or enter k-value.')
+                
+        #Check whether any one of the methods is selected to determine k-value
+        if determinek and not numofclus:
+            if determinek and not (minimumrange or maximumrange):
+                raise ValueError('Enter both minimumrange and maximumrange to determine k-value.')
+            logger.info('Determining k-value using ' + determinek)
+            kvalue = FEAT[determinek](data, minimumrange, maximumrange)
+        
+        #Check whether k-value is entered
+        if numofclus:
+            kvalue = numofclus
+        
+        #Cluster data using K-Means clustering
+        logger.info('Clustering the data')
+        cluster_data = kmeans_cluster(data,kvalue)
```

Comment:
Why cluster the data again if it's already been clustered when you want to determine k-value using an automated method?